includes/gpio.h

```diff
@@ -85,7 +91,7 @@ typedef struct  GPIO_config_s
 
 /* Note that to use GPIO you need to enable corresponding RCC */
 void    gpio_init(GPIO *GPIOx, GPIO_config *GPIO_init);
-void    gpio_select_alternate_function(GPIO *GPIOx, uint8_t pin, uint8_t function);
-void    gpio_set(GPIO *GPIOx, uint8_t pin, uint8_t value);
+void    gpio_select_alternate_function(GPIO_pin *gpio, uint8_t function);
+void    gpio_set(GPIO_pin *gpio, uint8_t value);
```

Comment:
Should be `const GPIO_pin *` here and above.

Reply:
Also consider pass by-copy instead of by-reference (`GPIO_pin gpio` instead of `GPIO_pin *gpio`).